engine/uState.ml

```diff
@@ -1062,6 +1080,21 @@ let update_sigma_univs uctx univs =
   in
   merge_seff eunivs eunivs.local
 
+let _reset_global_universes env uctx =
```

Comment:
this looks pretty similar to update_sigma_univs just above, except it has an extra `diff`. Should it instead replace update_sigma_univs?

Reply:
I don't think so, because update_sigma_univs is also used in other places without the diff part. This function is just the composition of update_sigma_univs with demote_global_univs, but contrarily to the latter it somewhat preserves the invariants of the ustate data type.

Reply:
(Also, this whole demote business shouldn't be here to begin with, it's just coq-elpi working around a deficient API in a similarly deficient way.)

Reply:
I will introduce this code in a future PR, for now I'm just changing the internals of the demote function.